plugin/harness/metadata.go

```diff
@@ -0,0 +1,53 @@
+package harness
+
+import (
+	"fmt"
+	"html/template"
+	"runtime"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Metadata implements a source url generator that uses a template engine
+// to generate url from give metadata.
+type Metadata struct {
+	tmpl    string
+	funcMap template.FuncMap
+}
+
+// NewMetadata creates a new source Generator.
+func NewMetadata(tmpl string, ref string) *Metadata {
```

Comment:
we can rename to `New` since the package is named metadata. e.g. go style guide prefers `metadata.New` as opposed to `metadata.NewMetadata`

Reply:
Package name is harness